perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -1228,7 +1246,7 @@ protected boolean hasCandidateExistingMember(Candidate candidate, RichMember ric
 	/**
 	 * This method run in separate transaction.
 	 */
-	public List<String> synchronizeGroup(PerunSession sess, Group group) throws InternalErrorException, MemberAlreadyRemovedException, AttributeNotExistsException, WrongAttributeAssignmentException, ExtSourceNotExistsException, WrongAttributeValueException, WrongReferenceAttributeValueException, GroupOperationsException, GroupNotExistsException {
+	public List<String> synchronizeGroup(PerunSession sess, Group group) throws InternalErrorException, MemberAlreadyRemovedException, AttributeNotExistsException, WrongAttributeAssignmentException, ExtSourceNotExistsException, WrongAttributeValueException, WrongReferenceAttributeValueException, GroupNotExistsException, AlreadyMemberException {
```

Comment:
Please check ``removeFormerMembersWhileSynchronization()`` used in this method. Its defined to throw ``AlreadyMemberException`` and ``MemberAlreadyRemovedException``, but they are not actually thrown and can be removed from it. Then it can be transitively removed from here and interface class too.